Source/WebCore/page/EventHandler.cpp

```diff
@@ -2333,10 +2333,30 @@ HandleUserInputEventResult EventHandler::handleMouseReleaseEvent(const PlatformM
         return true;
 
     bool swallowMouseUpEvent = !dispatchMouseEvent(eventNames().mouseupEvent, mouseEvent.protectedTargetNode().get(), m_clickCount, platformMouseEvent, FireMouseOverOut::No);
-    bool contextMenuEvent = platformMouseEvent.button() == MouseButton::Right;
 
-    auto nodeToClick = targetNodeForClickEvent(RefPtr { m_clickNode }.get(), mouseEvent.protectedTargetNode().get());
-    bool swallowClickEvent = m_clickCount > 0 && !contextMenuEvent && nodeToClick && !dispatchMouseEvent(eventNames().clickEvent, nodeToClick.get(), m_clickCount, platformMouseEvent, FireMouseOverOut::Yes);
+    bool swallowClickEvent = [&] {
+        if (!m_clickCount)
+            return false;
+
+        auto nodeToClick = targetNodeForClickEvent(RefPtr { m_clickNode }.get(), mouseEvent.protectedTargetNode().get());
+
+        if (!nodeToClick)
+            return false;
+
+        bool isPrimaryPointerButton = platformMouseEvent.button() == MouseButton::Left;
+
+        if (!isPrimaryPointerButton) {
+            if (!frame->settings().auxclickEventEnabled())
+                return false;
+
+            // The auxclick event should only be fired for the non-primary pointer buttons.
+            // In the case of right button, the auxclick event is dispatched after any contextmenu event.
+            return !dispatchMouseEvent(eventNames().auxclickEvent, nodeToClick.get(), m_clickCount, platformMouseEvent, FireMouseOverOut::Yes);
+        }
+
+        // The click event should only be fired for the primary pointer button.
+        return !dispatchMouseEvent(eventNames().clickEvent, nodeToClick.get(), m_clickCount, platformMouseEvent, FireMouseOverOut::Yes);
```

Comment:
Seems like we should just put the event name in a local variable and share this line of code.

Reply:
I had it like this originally, but once I added the new if condition branch for the feature flag, that branch no longer dispatches an event like this, so the logic would become slightly more complicated if the event name were to be put in a local variable.

Reply:
I understand, and yet I still think I might refactor it to share that line of code, given that we never dispatch 2 events, only 0 or 1. But I’m OK landing it like this.

Reply:
I'll make that change first, and also fix the tests too